x/shield/keeper/purchase.go

```diff
@@ -165,12 +194,22 @@ func (k Keeper) RemoveExpiredPurchases(ctx sdk.Context) {
 			for i := 0; i < len(purchaseList.Entries); {
 				entry := purchaseList.Entries[i]
 				// DeletionTime = ProtectionEndTime - ProtectionPeriod + ClaimPeriod + VotingPeriod
-				// If DeletionTime > Now, skip.
+				// If DeletionTime > currentBlockTime, skip.
 				if entry.ProtectionEndTime.Add(deletionPeriod).After(ctx.BlockTime()) {
 					i++
 					continue
 				}
-				// If DeletionTime <= Now, remove purchase and update shield.
+				// If previousBlockTime < ProtectionTime <= DeletionTime <= currentBlockTime,
+				// calculate remaining service fees to be distributed.
+				if entry.ProtectionEndTime.After(previousBlockTime) {
+					// Add purchaseServiceFees * (purchaseProtectionEndTime - previousBlockTime) / protectionPeriod.
+					serviceFees = serviceFees.Add(entry.ServiceFees.MulDec(
+						sdk.NewDec(int64(entry.ProtectionEndTime.Sub(previousBlockTime).Seconds())).Quo(
+							sdk.NewDec(int64(k.GetPoolParams(ctx).ProtectionPeriod.Seconds())))))
+					// Remove purchaseServiceFees from total service fees.
+					totalServiceFees = totalServiceFees.Sub(entry.ServiceFees)
+				}
+				// If DeletionTime <= currentBlockTime, remove purchase and update shield.
```

Comment:
This is not actually checked

Reply:
When DeletionTime > currentBlockTime we will call continue. If the code reaches this line, it means DeletionTime <= currentBlockTime.